src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorSpectrum.cs

```diff
@@ -0,0 +1,1810 @@
+			args.Handled = true;
+		}
+
+		private void OnInputTargetPointerReleased(object sender, PointerRoutedEventArgs args)
+		{
+			m_isPointerPressed = false;
+			m_shouldShowLargeSelection = false;
+
+			m_inputTarget.ReleasePointerCapture(args.Pointer);
+			UpdateVisualState(useTransitions: true);
+			UpdateEllipse();
+
+			args.Handled = true;
+		}
+
+		private void OnSelectionEllipseFlowDirectionChanged(DependencyObject o, DependencyProperty p)
+		{
+			UpdateEllipse();
+		}
+
+		private async void CreateBitmapsAndColorMap()
+		{
+			var layoutRoot = m_layoutRoot;
+			var sizingGrid = m_sizingGrid;
+			var inputTarget = m_inputTarget;
+			var spectrumRectangle = m_spectrumRectangle;
+			var spectrumEllipse = m_spectrumEllipse;
+			var spectrumOverlayRectangle = m_spectrumOverlayRectangle;
+			var spectrumOverlayEllipse = m_spectrumOverlayEllipse;
+
+			if (m_layoutRoot == null ||
+				m_sizingGrid == null ||
+				m_inputTarget == null ||
+				m_spectrumRectangle == null ||
+				m_spectrumEllipse == null ||
+				m_spectrumOverlayRectangle == null ||
+				m_spectrumOverlayEllipse == null ||
+				SharedHelpers.IsInDesignMode())
+			{
+				return;
+			}
+
+			double minDimension = Math.Min(layoutRoot.ActualWidth, layoutRoot.ActualHeight);
+
+			if (minDimension == 0)
+			{
+				return;
+			}
+
+			sizingGrid.Width = minDimension;
+			sizingGrid.Height = minDimension;
+
+			// Uno Doc: Slightly modified from WinUI
+			if (sizingGrid.Clip is RectangleGeometry clip)
+			{
+				clip.Rect = new Rect(0, 0, minDimension, minDimension);
+			}
+
+			inputTarget.Width = minDimension;
+			inputTarget.Height = minDimension;
+			spectrumRectangle.Width = minDimension;
+			spectrumRectangle.Height = minDimension;
+			spectrumEllipse.Width = minDimension;
+			spectrumEllipse.Height = minDimension;
+			spectrumOverlayRectangle.Width = minDimension;
+			spectrumOverlayRectangle.Height = minDimension;
+			spectrumOverlayEllipse.Width = minDimension;
+			spectrumOverlayEllipse.Height = minDimension;
+
+			Vector4 hsvColor = this.HsvColor;
+			int minHue = this.MinHue;
+			int maxHue = this.MaxHue;
+			int minSaturation = this.MinSaturation;
+			int maxSaturation = this.MaxSaturation;
+			int minValue = this.MinValue;
+			int maxValue = this.MaxValue;
+			ColorSpectrumShape shape = this.Shape;
+			ColorSpectrumComponents components = this.Components;
+
+			// If min >= max, then by convention, min is the only number that a property can have.
+			if (minHue >= maxHue)
+			{
+				maxHue = minHue;
+			}
+
+			if (minSaturation >= maxSaturation)
+			{
+				maxSaturation = minSaturation;
+			}
+
+			if (minValue >= maxValue)
+			{
+				maxValue = minValue;
+			}
+
+			Hsv hsv = new Hsv(Hsv.GetHue(hsvColor), Hsv.GetSaturation(hsvColor), Hsv.GetValue(hsvColor));
+
+			// The middle 4 are only needed and used in the case of hue as the third dimension.
+			// Saturation and luminosity need only a min and max.
+
+			byte[] test = Array.Empty<byte>();
```

Comment:
Codacy found an issue: [Remove the unused local variable 'test'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)